mitosheet/mitosheet/code_chunks/code_chunk.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# Copyright (c) Saga Inc.
+# Distributed under the terms of the GPL License.
+
+
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
+
+if TYPE_CHECKING:
+    from mitosheet.state import State
+else:
+    State = Any
+
+class CodeChunk:
+    """
+    A CodeChunk is the a abstract base class that can be inhereited from
+    to create a code generating object. For example, a AddColumnCodeChunk
+    with the correct parameters and states will generate code that adds
+    a column with a specific column header to a specific dataframe.
+
+    When a step is transpiled, it generates a list of code chunks. In turn,
+    we can optimize these code chunks to allow us to write the most minimal
+    amount of code. 
+    """
+
+    def __init__(self, 
+        prev_state: State,
+        post_state: State,
+        params: Dict[str, Any], 
+        execution_data: Optional[Dict[str, Any]]
+    ):
+        self.prev_state = prev_state
+        self.post_state = post_state
+        self.params = params
+        self.execution_data = execution_data
+
+    def get_display_name(self) -> str:
+        """Returns a short name to display for this CodeChunk"""
+        raise NotImplementedError('Implement in subclass')
```

Comment:
TODO: Understand where this gets displayed. Does the step history not update too? My assumption was that the step history taskpane was going to still show individual steps.

Reply:
Okay, yeah I think it is the case that we are updating the step history taskpane. That's the only reason that we would need the EmptyCodeCell CodeChunks. But I will see for sure when I test it

Reply:
> However, we need to decide what to do with the step list that is displayed in the frontend. Namely - how do we want to handle step checkout? Namely, because we no longer link the steps to specific code chunks, we don't have a notion of what step a code chunk corresponds to.

Would it be difficult to keep the step history as the unoptimized code? That would be my preference as I think it would be the most helpful version of the step history. And it seems like it wouldn't really incur much complexity at all.

Reply:
^ this was a good solution - like 10 lines of code! Nice :-) 

I just get the first code chunk for each step, and display that!

Reply:
^ this was a good solution - like 10 lines of code! Nice :-) 

I just get the first code chunk for each step, and display that!